test/testDB.js

```diff
@@ -0,0 +1,133 @@
+require("dotenv").config();
+const express = require("express");
+const { connect } = require("mongoose");
+const sinon = require("sinon");
+const { mockReq, mockRes } = require("sinon-express-mock");
+const request = require("supertest");
+const should = require('should');
+const authRouter = require("../api/auth/auth.router");
+
+
+const { checkAuthTokenMiddleware } = require("../middlewares/auth.middleware");
+const { avatarUploaderMiddleware } = require('../middlewares/fileUploader.middleware');
+const { uploadAvatarController } = require('../api/auth/auth.controller');
+
+describe('Connection mongo', () => {
+  it("Connection mongo test with valid URI", (done) => {
+    connect(process.env.DB_URI, { useNewUrlParser: true, useUnifiedTopology: true }, (err) => {
+      if (err) done(err);
+      done();
+    });
+  });
+})
+
+describe('Unit test for auth middleware', () => {
+  const argForMiddleware = () => {
+    const req = mockReq();
+    const res = mockRes({ params: {} });
+    const next = sinon.stub();
+  
+    res.status = function (params) {
+      this.params.status = params;
+      return this;
+    };
+    res.json = function (params) {
+      this.params.json = params;
+      return this;
+    };
+  
+    return [req, res, next];
+  };
+  
+  const validToken = () =>
+    "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVmNzRiNTE1ZTQ4ODRkMmM1NDcwMzA3NyIsImlhdCI6MTYwMTQ4NDEyOX0.YK-fzBeDVHckGo7NzuQTRxD1_9A2qjBViZqzabGqTlk";
+  
+  it("Check auth middleware with valid token", async () => {
+    return new Promise(async (resolve, reject) => {
```

Comment:
is it possible to write code without new Promise ?

Reply:
is it a question or a statement ?

Reply:
yes, this is possible